src/object/obj_class.c

```diff
@@ -675,6 +681,12 @@ oclass_ident2cl(daos_oclass_id_t oc_id)
 	return oc_ident_array[idx];
 }
 
+/**
+ * XXX: Function below should not look into the pre-registered list of
+ * class and now just generate it on the fly based on daos_obj_redun
+ * and customer nr_grp now encoded in the objid.
+ */
+
```

Comment:
All the functions below probably need to be reworked or just skipped. Except when the user specify a class manually, we shouldn't look at the list of pre-registered class otherwise.